clients/admin-ui/src/features/common/system-data-flow/DataFlowAccordionForm.tsx

```diff
@@ -0,0 +1,189 @@
+import {
+  AccordionButton,
+  AccordionIcon,
+  AccordionItem,
+  AccordionPanel,
+  Button,
+  ButtonGroup,
+  Flex,
+  Spacer,
+  Stack,
+  Tag,
+  Text,
+  useDisclosure,
+  useToast,
+} from "@fidesui/react";
+import { FormGuard } from "common/hooks/useIsAnyFormDirty";
+import { GearLightIcon } from "common/Icon";
+import QuestionTooltip from "common/QuestionTooltip";
+import { DataFlowSystemsDeleteTable } from "common/system-data-flow/DataFlowSystemsDeleteTable";
+import DataFlowSystemsModal from "common/system-data-flow/DataFlowSystemsModal";
+import { successToastParams } from "common/toast";
+import { Form, Formik } from "formik";
+import React, { useEffect, useMemo, useState } from "react";
+
+import {
+  useGetAllSystemsQuery,
+  useUpdateSystemMutation,
+} from "~/features/system";
+import { DataFlow, System } from "~/types/api";
+
+const defaultInitialValues = {
+  dataFlowSystems: [] as DataFlow[],
+};
+
+export type FormValues = typeof defaultInitialValues;
+
+type DataFlowAccordionItemProps = {
+  isIngress?: boolean;
+  system: System;
+  isSystemTab?: boolean;
+};
+
+export const DataFlowAccordionForm = ({
+  system,
+  isIngress,
+  isSystemTab,
+}: DataFlowAccordionItemProps) => {
+  const toast = useToast();
+  const flowType = isIngress ? "Source" : "Destination";
+  const pluralFlowType = `${flowType}s`;
+  const dataFlowSystemsModal = useDisclosure();
+  const [updateSystemMutationTrigger] = useUpdateSystemMutation();
+
+  const { data } = useGetAllSystemsQuery();
+  const systems = useMemo(() => data || [], [data]);
+
+  const initialDataFlows = useMemo(() => {
+    const dataFlows = isIngress ? system.ingress! : system.egress!;
+    const systemFidesKeys = systems ? systems.map((s) => s.fides_key) : [];
+
+    return dataFlows.filter((df) => systemFidesKeys.includes(df.fides_key));
+  }, [isIngress, system, systems]);
+
+  const [assignedDataFlow, setAssignedDataFlows] =
+    useState<DataFlow[]>(initialDataFlows);
+
+  useEffect(() => {
+    setAssignedDataFlows(initialDataFlows);
+  }, [initialDataFlows]);
+  const handleSubmit = async ({ dataFlowSystems }: FormValues) => {
+    const updatedSystem = {
+      ...system,
+      ingress: isIngress ? dataFlowSystems : system.ingress,
+      egress: !isIngress ? dataFlowSystems : system.egress,
+    };
+    await updateSystemMutationTrigger(updatedSystem);
+    toast(successToastParams(`${pluralFlowType} updated`));
+  };
+
+  return (
+    <AccordionItem>
+      <AccordionButton
+        height="68px"
+        paddingLeft={isSystemTab ? 6 : 2}
```

Comment:
The System configuration page the drawer require slightly different padding values